hbase-server/src/main/java/org/apache/hadoop/hbase/security/access/SnapshotScannerHDFSAclController.java

```diff
@@ -0,0 +1,667 @@
+        Set<TableName> skipTables = new HashSet<>();
+        List<byte[]> userEntries = SnapshotScannerHDFSAclStorage.getUserEntries(aclTable, userName);
+        for (byte[] entry : userEntries) {
+          if (!PermissionStorage.isNamespaceEntry(entry)
+              && !PermissionStorage.isGlobalEntry(entry)) {
+            skipTables.add(TableName.valueOf(entry));
+          }
+        }
+        hdfsAclHelper.revokeAcl(userPermission, new HashSet<>(), skipTables);
+      }
+      SnapshotScannerHDFSAclStorage.deleteUserNamespaceHdfsAcl(aclTable, userName, namespace);
+    }
+  }
+
+  private void removeUserTableHdfsAcl(Table aclTable, String userName, TableName tableName,
+      UserPermission userPermission) throws IOException {
+    if (SnapshotScannerHDFSAclStorage.hasUserTableHdfsAcl(aclTable, userName, tableName)) {
+      if (!SnapshotScannerHDFSAclStorage.hasUserGlobalHdfsAcl(aclTable, userName)
+          && !SnapshotScannerHDFSAclStorage.hasUserNamespaceHdfsAcl(aclTable, userName,
+            tableName.getNamespaceAsString())) {
+        // remove table acls
+        hdfsAclHelper.revokeAcl(userPermission, new HashSet<>(0), new HashSet<>(0));
+      }
+      SnapshotScannerHDFSAclStorage.deleteUserTableHdfsAcl(aclTable, userName, tableName);
+    }
+  }
+
+  private boolean containReadPermission(UserPermission userPermission) {
+    if (userPermission != null) {
+      return Arrays.stream(userPermission.getPermission().getActions())
+          .anyMatch(action -> action == Action.READ);
+    }
+    return false;
+  }
+
+  private UserPermission getUserGlobalPermission(Configuration conf, String userName)
+      throws IOException {
+    List<UserPermission> permissions = PermissionStorage.getUserPermissions(conf,
+      PermissionStorage.ACL_GLOBAL_NAME, null, null, userName, true);
+    if (permissions != null && permissions.size() > 0) {
+      return permissions.get(0);
+    }
+    return null;
+  }
+
+  private UserPermission getUserNamespacePermission(Configuration conf, String userName,
+      String namespace) throws IOException {
+    List<UserPermission> permissions =
+        PermissionStorage.getUserNamespacePermissions(conf, namespace, userName, true);
+    if (permissions != null && permissions.size() > 0) {
+      return permissions.get(0);
+    }
+    return null;
+  }
+
+  private UserPermission getUserTablePermission(Configuration conf, String userName,
+      TableName tableName) throws IOException {
+    List<UserPermission> permissions =
+        PermissionStorage.getUserTablePermissions(conf, tableName, null, null, userName, true);
+    if (permissions != null && permissions.size() > 0) {
+      return permissions.get(0);
+    }
+    return null;
+  }
+
+  /**
+   * Check if user global/namespace/table HDFS acls is already set to hfile
+   */
+  private boolean isHdfsAclSet(Table aclTable, String userName, Optional<String> namespace,
```

Comment:
The two Optional<String> argument should have the issue say: 

```
Reports any uses of java.util.Optional<T>, java.util.OptionalDouble, java.util.OptionalInt, java.util.OptionalLong or com.google.common.base.Optional as the type for a field or a parameter. Optional was designed to provide a limited mechanism for library method return types where there needed to be a clear way to represent "no result". Using a field with type java.util.Optional is also problematic if the class needs to be Serializable, which java.util.Optional is not.
```

Please don't use any Optional as the arguments.